pkg/scheduler/framework/plugins/dynamicresources/extended/cache.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package extended
+
+import (
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+	resourceapi "k8s.io/api/resource/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	klog "k8s.io/klog/v2"
+	fwk "k8s.io/kube-scheduler/framework"
+)
+
+// ExtendedResourceCache maintains a global cache of extended resource to device class mappings
+// that can be accessed by event handlers without requiring a scheduling cycle.
+type ExtendedResourceCache struct {
+	mutex sync.RWMutex
+	// mapping maps extended resource name to device class name
+	mapping    map[v1.ResourceName]string
+	draManager fwk.SharedDRAManager
+	logger     klog.Logger
+}
+
+// NewExtendedResourceCache creates a new ExtendedResourceCache instance.
+func NewExtendedResourceCache(draManager fwk.SharedDRAManager, logger klog.Logger) *ExtendedResourceCache {
+	cache := &ExtendedResourceCache{
+		mapping:    make(map[v1.ResourceName]string),
+		draManager: draManager,
+		logger:     logger,
+	}
+	logger.V(4).Info("Created extended resource cache")
+	return cache
+}
+
+// GetDeviceClass returns the device class name for the given extended resource name.
+// Returns the device class name and true if found, empty string and false otherwise.
+func (c *ExtendedResourceCache) GetDeviceClass(resourceName v1.ResourceName) (string, bool) {
+	c.mutex.RLock()
+	defer c.mutex.RUnlock()
+	deviceClass, exists := c.mapping[resourceName]
+	return deviceClass, exists
+}
+
+// GetAllMappings returns a copy of all current mappings.
+func (c *ExtendedResourceCache) GetAllMappings() map[v1.ResourceName]string {
```

Comment:
i think we can just use GetDeviceClass() to read directly from the cache, w/o needing to store a copy of it in noderesources and dynamicresource plugin.

Reply:
I remember adding this method for some reason while resolving some of the test case failures. But I got that working with this method now.